Dockerfile

```diff
@@ -12,17 +12,47 @@ ADD . /go-ethereum
 RUN cd /go-ethereum && go run build/ci.go install ./cmd/geth
 
 # Pull Geth into a second stage deploy alpine container
-FROM alpine:latest
+FROM alpine:3.16.0
+
+ARG BSC_USER=bsc
+ARG BSC_USER_UID=1000
+ARG BSC_USER_GID=1000
+
+ENV BSC_HOME=/bsc
+ENV HOME=${BSC_HOME}
+ENV DATA_DIR=/data
+
+ENV PACKAGES ca-certificates~=20211220 jq~=1.6 \
+  bash~=5.1.16-r2 bind-tools~=9.16.29-r0 tini~=0.19.0 \
+  grep~=3.7 curl==7.83.1-r2 sed~=4.8-r0
+
+RUN apk add --no-cache $PACKAGES \
+  && rm -rf /var/cache/apk/* \
+  && addgroup -g ${BSC_USER_GID} ${BSC_USER} \
+  && adduser -u ${BSC_USER_UID} -G ${BSC_USER} --shell /sbin/nologin --no-create-home -D ${BSC_USER} \
+  && addgroup ${BSC_USER} tty \
+  && sed -i -e "s/bin\/sh/bin\/bash/" /etc/passwd  
+
+RUN echo "[ ! -z \"\$TERM\" -a -r /etc/motd ] && cat /etc/motd" >> /etc/bash/bashrc
+
+WORKDIR ${BSC_HOME}
 
-RUN apk add --no-cache ca-certificates curl jq tini
 COPY --from=builder /go-ethereum/build/bin/geth /usr/local/bin/
 
-EXPOSE 8545 8546 8547 30303 30303/udp
-ENTRYPOINT ["geth"]
+COPY docker-entrypoint.sh ./
 
-# Add some metadata labels to help programatic image consumption
-ARG COMMIT=""
-ARG VERSION=""
-ARG BUILDNUM=""
+RUN chmod +x docker-entrypoint.sh \
+    && mkdir -p ${DATA_DIR} \
+    && chown -R ${BSC_USER_UID}:${BSC_USER_GID} ${BSC_HOME} ${DATA_DIR}
+
+VOLUME ${DATA_DIR}
+
+USER ${BSC_USER_UID}:${BSC_USER_GID}
+
+# mainnet rpc ws p2p
+EXPOSE 8545 8546 30311
```

Comment:
can we make the docker file simply, only expose the default ports

<img width="976" alt="image" src="https://user-images.githubusercontent.com/25412254/179468476-ce51025d-f399-4837-9de5-925dba9ae2f0.png">